src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginEntry/IFileOriginEntryViewModel.cs

```diff
@@ -0,0 +1,13 @@
+using System.Reactive;
+using ReactiveUI;
+
+namespace NexusMods.App.UI.Pages.ModLibrary.FileOriginEntry;
+
+public interface IFileOriginEntryViewModel : IViewModelInterface
+{
+    string Name { get; init;}
+    string Version { get; init; }
+    string Size { get; init; }    
+    string Date { get; init; }
```

Comment:
These shouldn't be strings. If they stay as strings, then they won't be sortable.

Reply:
We store versions as strings, and comparisons there should work.
I'm not really sure how to show "-" in case of unknown date though

Reply:
> We store versions as strings, and comparisons there should work.

Right.

>  I'm not really sure how to show "-" in case of unknown date though

Hijack how the dates are displayed. Use `Optional<DateTime>` or something, if it's `None` you display `-` or just display nothing.

Reply:
I'm doing a direct binding from the property to the DataGridTextColumn, there is no conversion step in the code behind or anything like that. I'd prefer not to do something like the LoadoutDataGrid where each column needs a custom view, vm, type, and an special definition.

Reply:
I added this to the things to to do https://github.com/Nexus-Mods/NexusMods.App/issues/1284

Reply:
You should be able to update the binding and change the string formatting or use a converter. If that doesn't work, you can always make the text columns into template columns.